File: src/Shared/Dtos/ScanDirectoryDto.cs
namespace Shared.Dtos;

/// <summary>
/// Data transfer object for scan directory configuration.
/// </summary>
public record ScanDirectoryDto
{
    public Guid Id { get; init; }
    public required string Path { get; init; }
    public bool IsEnabled { get; init; }
    public DateTime? LastScannedAt { get; init; }
    public DateTime CreatedAt { get; init; }
    public int FileCount { get; init; }
}

File: src/Web/src/app/models/scan-directory.dto.ts
/**
 * Data transfer object for scan directory configuration.
 * Matches backend: src/Shared/Dtos/ScanDirectoryDto.cs
 */
export interface ScanDirectoryDto {
  id: string;
  path: string;
  isEnabled: boolean;
  lastScannedAt: string | null;
  createdAt: string;
  fileCount: number;
}

/**
 * Request to create a new scan directory.
 * Matches backend: src/Shared/Requests/CreateScanDirectoryRequest.cs
 */
export interface CreateScanDirectoryRequest {
  path: string;
  isEnabled?: boolean;
}

/**
 * Request to update an existing scan directory.
 * Matches backend: src/Shared/Requests/UpdateScanDirectoryRequest.cs
 */
export interface UpdateScanDirectoryRequest {
  path?: string;
  isEnabled?: boolean;
}
